packages/flutter/lib/src/material/scaffold.dart

```diff
@@ -1783,6 +1784,20 @@ class Scaffold extends StatefulWidget {
   /// By default, the drag gesture is enabled.
   final bool endDrawerEnableOpenDragGesture;
 
+  /// Restoration ID to save and restore the state of the [Scaffold].
+  ///
+  /// If it is non-null, the scaffold will persist and restore whether the
+  /// [drawer] and [endDrawer] was open or closed.
+  ///
+  /// The state of this widget is persisted in a [RestorationBucket] claimed
+  /// from the surrounding [RestorationScope] using the provided restoration ID.
```

Comment:
This is not true, but it should be :)

Right now, it doesn't use the provided restoration ID and instead uses `${restorationId}_end_drawer` and `${restorationId}_drawer`.

I think you want ScaffoldState to use the RestorationMixin, which makes it claim a bucket using `restorationId`. Then you insert that Bucket into the widget hierarchy around the two DrawerController using a UnmanagedRestorationScope.

Reply:
I actually decided to pull out the restorable drawer open/close states into `ScaffoldState`, which I think is better because I worry about having two separate drawer open/close states in both the Scaffold and the Drawer widgets not aligning. PTAL and let me know what you think.

Reply:
Nice. That's even better!